lib/src/client/impl/exchange_impl.dart

```diff
@@ -22,7 +22,9 @@ class _ExchangeImpl implements Exchange {
   }
 
   void publish(Object message, String routingKey, { MessageProperties properties, bool mandatory : false, bool immediate : false}) {
-    if (type != ExchangeType.FANOUT && (routingKey == null || routingKey.isEmpty)) {
+    if(type is CustomExchangeType && (type as CustomExchangeType).routingKeyRequired && (routingKey == null || routingKey.isEmpty)){
```

Comment:
Sorry, I missed this earlier. So, I took out that routeKeyRequired stuff; it seemed to make more sense to loosen the route key requirements on custom exchanges since they could have an number of declaration requirements.